### ISSUE TITLE
Error with variables in templates if the template is inserted from not the beginning of the line.

### ISSUE THREAD

**Issue by VASilaev** [dbeaver/dbeaver#8347]
#### System information: 
- Windows 10
- 7.0.2

#### Connection specification:
- postgresSQL

#### Describe the problem you're observing:
Error with variables in templates if the template is not inserted from the beginning of the line.

#### Steps to reproduce, if exist:
For example, there is a template:
```
begin
${cursor};
end;
If you insert it with an indent of 4 spaces the following text will be inserted
    begin
    sor;
    end;
```
in other words, the Template breaks, and it gets even worse when there are several variables.

#### Include any warning/errors/backtraces from the logs

to solve this problem, I suggest the following code in the file \dbeaver-devel\plugins\org.jkiss.dbeaver.ui.editors.sql\src\org\jkiss\dbeaver\ui\editors\sql\templates\SQLContext.java:  private void formatTemplate(TemplateBuffer buffer) {

    private void formatTemplate(TemplateBuffer buffer) {
        TemplateVariable[] variables= buffer.getVariables();
        final String indentation = getIndentation();
        String content = buffer.getString();
        if (!indentation.isEmpty() && content.indexOf('\n') != -1) {
            StringBuilder result = new StringBuilder();
            
            //skolko mesta nugno
            int nCountPlace = 0;            
            for (int i = 0; i < variables.length; i++) {
              nCountPlace = nCountPlace + variables[i].getOffsets().length;
            };
            
            //zapolnyem masssiv
            int aVarOffset[][] = new int[nCountPlace][2];
            nCountPlace = 0;
            for (int i = 0; i < variables.length; i++) {
              int[] aOffsets = variables[i].getOffsets();
              for (int j = 0; j < aOffsets.length; j++) {
                aVarOffset[nCountPlace][0] = aOffsets[j];
                aVarOffset[nCountPlace][1] = i;
                nCountPlace = nCountPlace + 1; 
              };
            };
            
            //sort
            for (int i = 0; i < nCountPlace - 1; i++) {
              for (int j = i+1; j < nCountPlace; j++) {
                if (aVarOffset[i][0] > aVarOffset[j][0]) {
                  int tmp = aVarOffset[i][0];
                  aVarOffset[i][0] = aVarOffset[j][0];
                  aVarOffset[j][0] = tmp;
                  tmp = aVarOffset[i][1];
                  aVarOffset[i][1] = aVarOffset[j][1];
                  aVarOffset[j][1] = tmp;
                };
              };
            };

            int iPlace = 0;
            int iSpaceLen = indentation.length();
            
            for (int i = 0; i < content.length(); i++) {
                char c = content.charAt(i);
                result.append(c);
                if (c == '\n') {
                  //Propuskaem 
                  while (iPlace < nCountPlace && i > aVarOffset[iPlace][0]) iPlace++;
                  //Dvigaem
                  for (int j = iPlace; j < nCountPlace; j++)  aVarOffset[j][0] = aVarOffset[j][0] + iSpaceLen;
                  result.append(indentation);
                }
            }

            //sobiraem novye offsety
            for (int i = 0; i < variables.length; i++) {
              int[] aOffsets = variables[i].getOffsets();
              int iInd = 0;
              for (int j = 0; j < nCountPlace; j++) {
                if (aVarOffset[j][1] == i) {
                  aOffsets[iInd] = aVarOffset[j][0];
                  iInd++;
                };
              };
            };
            
            
            buffer.setContent(result.toString(), variables);
        }
    }

**Comment by kseniiaguzeeva** [comment]
Thank you for the bug report.

**Comment by serge-rider** [comment]
Fixed by PR #8374

**Comment by kseniiaguzeeva** [comment]
verified
